### PR TITLE
HBASE-24485 Backport to branch-1 HBASE-17738 BucketCache startup is slow

### PR DESCRIPTION
**Pull request by thangTang**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m  7s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m  3s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   1m 10s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   2m 24s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 28s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  1s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   1m  7s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   2m 53s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 15s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 40s |  root in the patch failed.  |
| -1 :x: |  compile  |   0m 16s |  hbase-common in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  compile  |   0m 19s |  hbase-server in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  javac  |   0m 16s |  hbase-common in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  javac  |   0m 19s |  hbase-server in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  compile  |   0m 19s |  hbase-common in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  compile  |   0m 22s |  hbase-server in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  javac  |   0m 19s |  hbase-common in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  javac  |   0m 22s |  hbase-server in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  checkstyle  |   0m 29s |  hbase-common: The patch generated 7 new + 41 unchanged - 3 fixed = 48 total (was 44)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  shadedjars  |   0m 41s |  patch has 14 errors when building our shaded downstream artifacts.  |
| -1 :x: |  hadoopcheck  |   0m 35s |  The patch causes 14 errors with Hadoop v2.8.5.  |
| -1 :x: |  hadoopcheck  |   1m 23s |  The patch causes 14 errors with Hadoop v2.9.2.  |
| -1 :x: |  javadoc  |   0m 17s |  hbase-common-jdk1.8.0_252 with JDK v1.8.0_252 generated 1 new + 1 unchanged - 0 fixed = 2 total (was 1)  |
| -1 :x: |  javadoc  |   0m 35s |  hbase-server-jdk1.8.0_252 with JDK v1.8.0_252 generated 1 new + 8 unchanged - 0 fixed = 9 total (was 8)  |
| -1 :x: |  javadoc  |   0m 24s |  hbase-common-jdk1.7.0_262 with JDK v1.7.0_262 generated 1 new + 5 unchanged - 0 fixed = 6 total (was 5)  |
| -1 :x: |  javadoc  |   0m 20s |  hbase-server in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  findbugs  |   0m 17s |  hbase-common in the patch failed.  |
| -1 :x: |  findbugs  |   0m 22s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 17s |  hbase-common in the patch failed.  |
| -1 :x: |  unit  |   0m 22s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 21s |  The patch does not generate ASF License warnings.  |
|  |   |  37m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.10 Server=19.03.10 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1823 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2def75b154e9 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1823/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4096925 |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-common-jdk1.8.0_252.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-server-jdk1.8.0_252.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-common-jdk1.8.0_252.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-server-jdk1.8.0_252.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-common-jdk1.7.0_262.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-server-jdk1.7.0_262.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-common-jdk1.7.0_262.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-compile-hbase-server-jdk1.7.0_262.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/diff-checkstyle-hbase-common.txt |
| shadedjars | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-shadedjars.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-javac-2.8.5.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-javac-2.9.2.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/diff-javadoc-javadoc-hbase-common-jdk1.8.0_252.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/diff-javadoc-javadoc-hbase-server-jdk1.8.0_252.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/diff-javadoc-javadoc-hbase-common-jdk1.7.0_262.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-javadoc-hbase-server-jdk1.7.0_262.txt |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-findbugs-hbase-common.txt |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-findbugs-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-unit-hbase-common.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/testReport/ |
| Max. process+thread count | 77 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 49s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 20s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m 12s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   1m 18s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   2m 36s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 45s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  9s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   1m 13s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   3m 14s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 43s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 45s |  root in the patch failed.  |
| -1 :x: |  compile  |   0m 19s |  hbase-common in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  compile  |   0m 30s |  hbase-server in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  javac  |   0m 19s |  hbase-common in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  javac  |   0m 30s |  hbase-server in the patch failed with JDK v1.8.0_252.  |
| -1 :x: |  compile  |   0m 22s |  hbase-common in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  compile  |   0m 35s |  hbase-server in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  javac  |   0m 22s |  hbase-common in the patch failed with JDK v1.7.0_262.  |
| -1 :x: |  javac  |   0m 35s |  hbase-server in the patch failed with JDK v1.7.0_262.  |
| +1 :green_heart: |  checkstyle  |   0m 31s |  hbase-common: The patch generated 0 new + 1 unchanged - 3 fixed = 1 total (was 4)  |
| -1 :x: |  checkstyle  |   1m 51s |  hbase-server: The patch generated 1 new + 4 unchanged - 0 fixed = 5 total (was 4)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| -1 :x: |  shadedjars  |   0m 52s |  patch has 20 errors when building our shaded downstream artifacts.  |
| -1 :x: |  hadoopcheck  |   0m 41s |  The patch causes 20 errors with Hadoop v2.8.5.  |
| -1 :x: |  hadoopcheck  |   1m 37s |  The patch causes 20 errors with Hadoop v2.9.2.  |
| -1 :x: |  javadoc  |   0m 37s |  hbase-server-jdk1.8.0_252 with JDK v1.8.0_252 generated 1 new + 8 unchanged - 0 fixed = 9 total (was 8)  |
| -1 :x: |  javadoc  |   0m 46s |  hbase-server-jdk1.7.0_262 with JDK v1.7.0_262 generated 1 new + 8 unchanged - 0 fixed = 9 total (was 8)  |
| -1 :x: |  findbugs  |   0m 22s |  hbase-common in the patch failed.  |
| -1 :x: |  findbugs  |   0m 34s |  hbase-server in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 24s |  hbase-common in the patch failed.  |
| -1 :x: |  unit  |   0m 34s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 24s |  The patch does not generate ASF License warnings.  |
|  |   |  42m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.10 Server=19.03.10 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1823 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 92765d1fd068 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1823/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4096925 |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
| mvninstall | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-common-jdk1.8.0_252.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-server-jdk1.8.0_252.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-common-jdk1.8.0_252.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-server-jdk1.8.0_252.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-common-jdk1.7.0_262.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-server-jdk1.7.0_262.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-common-jdk1.7.0_262.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-compile-hbase-server-jdk1.7.0_262.txt |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/diff-checkstyle-hbase-server.txt |
| shadedjars | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-shadedjars.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-javac-2.8.5.txt |
| hadoopcheck | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-javac-2.9.2.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/diff-javadoc-javadoc-hbase-server-jdk1.8.0_252.txt |
| javadoc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/diff-javadoc-javadoc-hbase-server-jdk1.7.0_262.txt |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-findbugs-hbase-common.txt |
| findbugs | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-findbugs-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-unit-hbase-common.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/testReport/ |
| Max. process+thread count | 77 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 12s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m  5s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   1m 15s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   2m 23s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   2m 59s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 59s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   1m  3s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   2m 38s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 54s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  5s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  javac  |   1m  5s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  hbase-common: The patch generated 0 new + 1 unchanged - 3 fixed = 1 total (was 4)  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  The patch passed checkstyle in hbase-server  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 36s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   1m  5s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  findbugs  |   4m  1s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 45s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 122m 53s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 56s |  The patch does not generate ASF License warnings.  |
|  |   | 174m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.11 Server=19.03.11 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1823 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 0dc6d451dcc2 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1823/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4096925 |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/3/testReport/ |
| Max. process+thread count | 4428 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java [hunk 1]
```diff
@@ -102,8 +170,8 @@ public int getMultiple(long start, int len, byte[] dstArray, int dstOffset) {
 
   private final static Visitor GET_MULTIPLE_VISTOR = new Visitor() {
     @Override
-    public void visit(ByteBuffer bb, byte[] array, int arrayIdx, int len) {
-      bb.get(array, arrayIdx, len);
+    public void visit(ByteBuffer bb, int pos, byte[] array, int arrayIdx, int len) {
+      ByteBufferUtils.copyFromBufferToArray(array, bb, pos, arrayIdx, len);
```
Comment (infraio) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java:174]: Why change this? Seems this not belong to HBASE-17738?

File: hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java [hunk 2]
```diff
@@ -142,11 +210,12 @@ public void visit(ByteBuffer bb, byte[] array, int arrayIdx, int len) {
      * bytes from the buffer to the destination array, else if it is a write
      * action, we will transfer the bytes from the source array to the buffer
      * @param bb byte buffer
+     * @param pos Start position in ByteBuffer
      * @param array a source or destination byte array
      * @param arrayOffset offset of the byte array
      * @param len read/write length
      */
-    void visit(ByteBuffer bb, byte[] array, int arrayOffset, int len);
+    void visit(ByteBuffer bb, int pos, byte[] array, int arrayOffset, int len);
```
Comment (infraio) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java:218]: Ditto.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java [hunk 3]
```diff
@@ -169,34 +238,29 @@ void multiple(long start, int len, byte[] array, int arrayOffset, Visitor visito
     assert startBuffer >= 0 && startBuffer < bufferCount;
     assert (endBuffer >= 0 && endBuffer < bufferCount)
         || (endBuffer == bufferCount && endOffset == 0);
-    if (startBuffer >= locks.length || startBuffer < 0) {
+    if (startBuffer >= buffers.length || startBuffer < 0) {
```
Comment (infraio) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java:241]: Ditto.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java (hunk 1 above)
Comment (thangTang) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java:174]: > Why change this? Seems this not belong to HBASE-17738?

Also removed some Unnecessary locks related to [HBASE-15785](https://issues.apache.org/jira/browse/HBASE-15785)

Comment (thangTang) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java:174]: Should i backport this patch first?

Comment (infraio) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/util/ByteBufferArray.java:174]: This backport have to depend on that patch? Should be another backport if it is necessary.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 28s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 17s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m 12s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   1m 16s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   2m 31s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 30s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  7s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   1m 16s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   3m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 29s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  7s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javac  |   1m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 12s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  javac  |   1m 12s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 31s |  hbase-common: The patch generated 0 new + 1 unchanged - 1 fixed = 1 total (was 2)  |
| +1 :green_heart: |  checkstyle  |   1m 49s |  The patch passed checkstyle in hbase-server  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 11s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 18s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   1m  2s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   1m 10s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  findbugs  |   4m 51s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 51s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 162m 43s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 53s |  The patch does not generate ASF License warnings.  |
|  |   | 221m 12s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.11 Server=19.03.11 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1823 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux fa0a5edcab6f 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1823/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4118383 |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/4/testReport/ |
| Max. process+thread count | 5068 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1823/4/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by thangTang** [comment]
solved conflicts and repushed.

**Pull request merged**